### PR TITLE
some coverage improvements

### PR DESCRIPTION
**Pull request by naylor-b**
### Summary

- got rid of some dead code
-  modified behavior of _no_check_partials so it's ignored if we're running under CI because we were missing coverage of the compute_partials/linearize methods of some of our built-in components.
- fixed a couple of 'casting away complex' warnings

### CODE REVIEW

File: .github/workflows/openmdao_test_workflow.yml [hunk 1]
```diff
@@ -458,7 +461,7 @@ jobs:
           echo "============================================================="
           copy .coveragerc $HOME
           cd $HOME
-          testflo -n 2 openmdao --timeout=120 --show_skipped --coverage  --coverpkg openmdao --durations=20
+          testflo -n 1 openmdao --timeout=120 --show_skipped --coverage  --coverpkg openmdao --durations=20
```
Comment (swryan) [inline .github/workflows/openmdao_test_workflow.yml:464]: is there an issue with `-n 2` on Windows?  might want a comment here to document why...

File: openmdao/components/tests/test_exec_comp.py [hunk 2]
```diff
@@ -1685,7 +1686,10 @@ def test_register_check_partials_safe(self):
             assert_near_equal(p['comp.area_square'], np.ones(size) * 9., 1e-11)
 
             data = p.check_partials(out_stream=None)
-            self.assertEqual(list(data), [])
+            if env_truthy('GITHUB_ACTION') or env_truthy('TRAVIS'):
```
Comment (swryan) [inline openmdao/components/tests/test_exec_comp.py:1689]: I think we can just use the environment variable `CI` here..  it's boolean and available on GH Actions, Travis-CI and CircleCI:
https://docs.github.com/en/actions/learn-github-actions/environment-variables
https://docs.travis-ci.com/user/environment-variables
https://circleci.com/docs/env-vars

File: openmdao/core/problem.py [hunk 3]
```diff
@@ -1268,8 +1268,11 @@ def check_partials(self, out_stream=_DEFAULT_OUT_STREAM, includes=None, excludes
         excludes = [excludes] if isinstance(excludes, str) else excludes
 
         comps = []
+        under_CI = env_truthy('GITHUB_ACTION') or env_truthy('TRAVIS')
```
Comment (swryan) [inline openmdao/core/problem.py:1271]: I think we can just use the environment variable `CI` here..  it's boolean and available on GH Actions, Travis-CI and CircleCI:
https://docs.github.com/en/actions/learn-github-actions/environment-variables
https://docs.travis-ci.com/user/environment-variables
https://circleci.com/docs/env-vars

File: openmdao/core/tests/test_check_derivs.py [hunk 4]
```diff
@@ -240,7 +241,10 @@ def test_component_no_check_partials(self):
         # disable partials on comp2
         #
         comp2._no_check_partials = True
-        data = prob.check_partials(out_stream=None)
+
+        # Make check_partials think we're not on CI so we'll get the expected  non-CI behavior
+        with set_env_vars_context(GITHUB_ACTION='0', TRAVIS='0'):
```
Comment (swryan) [inline openmdao/core/tests/test_check_derivs.py:246]: I think we can just use the environment variable `CI` here..  it's boolean and available on GH Actions, Travis-CI and CircleCI:
https://docs.github.com/en/actions/learn-github-actions/environment-variables
https://docs.travis-ci.com/user/environment-variables
https://circleci.com/docs/env-vars

**Review by swryan — CHANGES_REQUESTED** [review]

File: openmdao/components/tests/test_exec_comp.py (hunk 2 above)
Comment (naylor-b) [reply openmdao/components/tests/test_exec_comp.py]: fixed

File: openmdao/core/problem.py (hunk 3 above)
Comment (naylor-b) [reply openmdao/core/problem.py]: fixed

**Review by swryan — APPROVED** [review]

**Review by Kenneth-T-Moore — APPROVED** [review]

**Pull request merged**